src/services/request-service.ts

```diff
@@ -0,0 +1,85 @@
+import Promise = require('bluebird');
+import r = require('rethinkdb');
+import uuid = require('uuid');
+import EmailService = require('./email-service');
+import nodemailer = require('nodemailer');
+import q = require('../dbutils/query');
+import models = require('../models/models');
+import errors = require('../errors/errors');
+import config = require('../config');
+
+
+module RequestService {
+    var db = "froyo";
+    var table = "requests";
+
+    export function requestExists(userID:string, carpoolID:string) : Promise<boolean> {
+        return q.run(
+                     r.db(db)
+                    .table(table)
+                    .filter({userID: userID,carpoolID: carpoolID})
+                    .isEmpty()
+                    .not()
+                )().then( (result) => {
+                    return result;
+                })
+    }
+
+    export function createRequest(userID:string, carpoolID:string) : Promise<boolean> {
+        
+        var insertRequest = r.db(db).table(table).insert({userID,carpoolID});
+        return requestExists(userID, carpoolID).then( (result) => {
```

Comment:
This can be written like following. It is easier to read, I think.

```
return requestExists(userID, carpoolID)
   .then((result) =>  {
     if (result) {
       throw new errors.CarpoolRequestConflict();
     }
   })
   .then(q.run(insertRequest))
```